modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4WriteThrottlingHandler.java

```diff
@@ -29,12 +32,17 @@
     private final Queue<WriteOperation> queuedWrites = new ArrayDeque<>();
 
     private WriteOperation currentWrite;
+    private ThreadPool threadPool;
 
-    public Netty4WriteThrottlingHandler() {}
+    public Netty4WriteThrottlingHandler(ThreadPool threadPool) {
```

Comment:
I'd rather capture just the `ThreadContext` itself here, not the whole `ThreadPool`.